frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/BranchNamesCompletion.java

```diff
@@ -0,0 +1,92 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.codeInsight.completion.InsertHandler;
+import com.intellij.codeInsight.completion.InsertionContext;
+import com.intellij.codeInsight.completion.PlainPrefixMatcher;
+import com.intellij.codeInsight.lookup.CharFilter;
+import com.intellij.codeInsight.lookup.LookupElement;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.util.textCompletion.DefaultTextCompletionValueDescriptor;
+import com.intellij.util.textCompletion.TextCompletionProvider;
+import com.intellij.util.textCompletion.TextCompletionValueDescriptor;
+import io.vavr.collection.List;
+import lombok.val;
+
+// TODO (#1604): remove this class and replace with com.intellij.util.textCompletion.TextCompletionProviderBase
```

Comment:
I'm assuming that's more or less 1:1 copierd from the original source? 😅

Reply:
yes! and simplified for our single purpose